crates/autopilot/src/run_loop.rs

```diff
@@ -76,9 +75,10 @@ pub struct RunLoop {
     /// the most recent data available.
     maintenance: Arc<Maintenance>,
     competition_updates_sender: tokio::sync::mpsc::UnboundedSender<()>,
+    auction_mechanism: T,
```

Comment:
Storing the `T` directly is usually not worth the hassle. As you can see you now have to put a bunch of constraints on the `RunLoop` type.
Instead storing `Box<dyn AuctionMechanism>` is usually a lot more convenient. And the indirection of chasing a pointer isn't really a big issue anyway.